core/src/test/java/org/infinispan/container/versioning/AbstractClusteredWriteSkewTest.java

```diff
@@ -301,19 +301,22 @@ private void doTestWriteSkewWithPassivation(boolean executeOnPrimaryOwner, Opera
       }
 
       DataContainer dataContainer = TestingUtil.extractComponent(primaryOwner, DataContainer.class);
-      MagicKey[] keys = new MagicKey[MAX_ENTRIES];
+      int retryCount = 5;
       int insertCount = 10;
-      for (int i = 0; i < insertCount; ++i) {
-         for (int j = 0; j < MAX_ENTRIES; ++j) {
-            MagicKey tempKey = keys[j];
-            if (tempKey == null) {
-               tempKey = new MagicKey("other-key-" + j, primaryOwner);
-               keys[j] = tempKey;
+      for (int retry = 0; retry < 5; ++retry) {
```

Comment:
Yeah tbh the more I think about it we would probably really want this (for non test code). The reason being is that when we do a put we first do a read before hand, so we can possibly do some verification checks of version and other things before actually doing the real put. This would lead to entries being promoted much more quickly in the eviction structure since there would be multiple hits.